apps/browser/qml/pages/components/BookmarkItem.qml

```diff
@@ -120,4 +123,34 @@ ListItem {
             }
         }
     }
+
+    Component {
+        id: desktopBookmarkWriter
+        DesktopBookmarkWriter {
+            onSaved: destroy()
+        }
+    }
+
+    Component {
+        id: editDialog
+        BookmarkEditDialog {
+            onAccepted: bookmarkModel.edit(index, editedUrl, editedTitle)
+        }
+    }
+
+    Component {
+        id: addToLauncherDialog
+        BookmarkEditDialog {
```

Comment:
This could even be wrapped into a component AddToAppGridDialog and then FavoriteGrid updated accordingly. Now code is copy-paste.